File: app/src/main/java/com/coder/ffmpegtest/service/FFmpegCommandService.java
package com.coder.ffmpegtest.service;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

import com.coder.ffmpeg.jni.FFmpegCommand;

import java.io.File;
import java.util.Locale;

import androidx.annotation.Nullable;

/**
 * @author: AnJoiner
 * @datetime: 20-6-28
 */
public class FFmpegCommandService extends Service {
    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
//        String videoPath =
//                Environment.getExternalStorageDirectory().getPath() + File.separator +
//                        "DCIM" + File.separator + "test.mp4";
        String videoPath = new File(getExternalCacheDir(), "test.mp4").getAbsolutePath();

        String output = new File(getExternalCacheDir(), "output.yuv").getAbsolutePath();

        String cmd = "ffmpeg -y -i %s -an -c:v rawvideo -pixel_format yuv420p %s";
        final String result = String.format(Locale.CHINA, cmd, videoPath, output);
        final String[] strings = result.split(" ");

        FFmpegCommand.runCmd(strings);

        return super.onStartCommand(intent, flags, startId);
    }
}

File: app/src/main/java/com/coder/ffmpegtest/service/FFmpegCommandService.kt
package com.coder.ffmpegtest.service

import android.app.IntentService
import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.util.Log
import com.coder.ffmpeg.annotation.MediaAttribute
import com.coder.ffmpeg.call.CommonCallBack
import com.coder.ffmpeg.jni.FFmpegCommand
import com.coder.ffmpeg.utils.CommandParams
import com.coder.ffmpegtest.utils.FileUtils
import com.coder.ffmpegtest.utils.ToastUtils
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import java.io.File
import java.util.*

/**
 * @author: AnJoiner
 * @datetime: 20-6-28
 */
class FFmpegCommandService : IntentService("") {
    companion object{
        val TAG = "FFmpegCommandService"
    }
    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        Log.e(TAG,"onStartCommand")
        FileUtils.copy2Memory(this, "test.mp4")
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onDestroy() {
        Log.e(TAG,"onDestroy")
        super.onDestroy()
    }

    override fun onHandleIntent(intent: Intent?) {
        val videoPath = File(externalCacheDir, "test.mp4").absolutePath
        val output = File(externalCacheDir, "leak.avi").absolutePath
//        val cmd = "ffmpeg -y -i %s -b:v 600k %s"
        val command = CommandParams()
            .append("-i")
            .append(videoPath)
            .append("-b:v")
            .append("600k")
            .append(output)
            .get()
        FFmpegCommand.runCmd(command, callback("测试内存抖动", output))
    }

    private fun callback(msg: String, targetPath: String?): CommonCallBack? {
        return object : CommonCallBack() {
            override fun onStart() {
                Log.d("FFmpegCmd", "onStart")
            }

            override fun onComplete() {
                Log.d("FFmpegCmd", "onComplete")

            }

            override fun onCancel() {
                Log.d("FFmpegCmd", "Cancel")
            }

            override fun onProgress(progress: Int, pts: Long) {
                Log.d("FFmpegCmd", progress.toString() + "")
            }

            override fun onError(errorCode: Int, errorMsg: String?) {
                Log.d("FFmpegCmd", errorMsg+"")
            }
        }
    }
}